subsys/net/lib/azure_fota/azure_fota.c

```diff
@@ -227,6 +227,14 @@ static bool parse_reported_status(const char *msg)
 	LOG_DBG("Currently reported 'fwUpdateStatus' in device twin: %s",
 		log_strdup(fw_status_obj->valuestring));
 
+	/* "current" status indicates no FOTA was in progress, only allow
+	 * updates with this status to be applied.
+	 */
+	if (strcmp("current", fw_status_obj->valuestring) != 0) {
+		LOG_DBG("Firmware update already in progress.");
+		goto clean_exit;
+	}
```

Comment:
Sorry about the very delayed response, this one completely slipped through the cracks.
I think FOTA will still be started with this patch. The difference will be that the device does not report back "current" status, but retains the previous status. If there's information about a pending update in the desired object, it will still be picked up here:
https://github.com/nrfconnect/sdk-nrf/blob/314ccaf0d90e0fd81562058544d42b51262f3109/subsys/net/lib/azure_fota/azure_fota.c#L501

I'm wondering if this could solved by adding something like a job ID to the FOTA information in the desired object, and report the same ID back once the FOTA is started. Then we can add some device side logic to not restart a download with the same ID, but at the same time allow retries with the same firmware version. Otherwise I'm not sure how to solve this without having the cloud side remove the information in the desired object once a FOTA download is started.